src/system/SystemTimer.h

```diff
@@ -43,47 +46,111 @@
 namespace chip {
 namespace System {
 
-class Layer;
+namespace Timers {
+
+typedef void (*OnCompleteFunct)(Layer * aLayer, void * appState, CHIP_ERROR aError);
+
+} // namespace Timers
+
+#if CHIP_SYSTEM_CONFIG_NUM_TIMERS
```

Comment:
CHIP_SYSTEM_CONFIG_NUM_TIMERS is used to define the timer pool size, it is strange to be reused as the feature flag. I think timer support is mandatory for all chip platforms

Reply:
All platforms need to support the timer API in `System::Layer`, but they no longer need to use the `System::Timer` pool to do it. For instance, Darwin can soon just use its native dispatch queues.

I intend to update the libevent example implementation to not use the pool, but first need to fix a bunch of tests that are calling into already-destructed `System::Layer` due to undefined global destruction order.

Reply:
Agreed, my point is CHIP_SYSTEM_CONFIG_NUM_TIMERS should not be reused as the feature flag, for exmaple, if it is defined as -1, all timer APIs  are kept in the translation unit, even it is invalid length of pool size